### PR TITLE
[Test] Integration test for community.py

### LINKED ISSUE #152: [Test] Integration Tests for lib/analysis/author/
**Description**
- [x] community.py
- [ ] ranking.py
- [x] edge_list.py
- [x] time_statistics.py
- [x] wh_table.py
- [x] curve_fitting.py

In graph/
- [x] bipartite.py
- [x] generate.py
- [x] interaction.py

### PR DESCRIPTION
**Pull request by hrishikesh-dahiya**
## Description
 - Integration test for community.py
 - Two methods of community.py moved to deprecated/
 - Fixes #152 

## Motivation
 - To increase code coverage
 
 ## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checks
 - [x] Squash your commits down to one, or a few, discreet changesets before submitting a pull request
 - [x] Read the [contributing guide](https://github.com/DeveloperCAP/MLCAT/blob/development/.github/CONTRIBUTING.md)
 - [x] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [ ] If possible, include an example in the description, that could help all readers (including project members) to get a better picture of the change
 - [x] Avoid other runtime dependencies unless absolutely necessary 
 - [x] Rebase your commit history so that each commit is meaningful and states the intention behind the change 
 - [x] Ensure that your pull request follows the [style guidelines](https://github.com/DeveloperCAP/MLCAT/wiki/Coding-Guidelines)
 - [x] At least one commit should mention `Fixes #152` _if relevant_

## Notify reviewers
@prasadtalasila

### CODE REVIEW

**Comment by prasadtalasila** [comment]
@hrishikesh-dahiya please resolve the merge conflict.

**Comment by hrishikesh-dahiya** [comment]
@prasadtalasila there are no merge conflicts in this PR, though travis CI is erroring in installation of igraph dependency

File: etc/install_deps.sh [hunk 1]
```diff
@@ -1,20 +1,14 @@
 #!/bin/bash
```
Comment (prasadtalasila) [inline etc/install_deps.sh]: given that this file is being tailored for travis deployment, it is better to name this as `travis_installs.sh` and keep the original `install_deps.sh` for local installations.

File: etc/igraph.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+wget http://switch.dl.sourceforge.net/sourceforge/igraph/igraph-0.5.4.tar.gz
```
Comment (prasadtalasila) [inline etc/igraph.sh]: We can probably invoke this script inside `travis_installs.sh` and `install_deps.sh`

**Review by prasadtalasila — CHANGES_REQUESTED** [review]

**Comment by prasadtalasila** [comment]
@hrishikesh-dahiya good work on this PR.

**Review by prasadtalasila — APPROVED** [review]

**Pull request merged**